homeassistant/components/light/xiaomi_miio.py

```diff
@@ -351,13 +417,24 @@ def async_update(self):
             _LOGGER.debug("Got new state: %s", state)
 
             self._state = state.is_on
-            self._brightness = ceil((255/100.0) * state.brightness)
+            self._brightness = ceil((255 / 100.0) * state.brightness)
             self._color_temp = self.translate(
                 state.color_temperature,
                 CCT_MIN, CCT_MAX,
                 self.max_mireds, self.min_mireds)
 
+            delayed_turn_off = self.delayed_turn_off_timestamp(
+                state.delay_off_countdown,
+                dt.utcnow(),
+                self._state_attrs[ATTR_DELAYED_TURN_OFF])
+
+            self._state_attrs.update({
+                ATTR_SCENE: state.scene,
+                ATTR_DELAYED_TURN_OFF: delayed_turn_off,
```

Comment:
It's still not clear to me what we're storing here? Is this a timestamp? If so, why are we recalculating the timestamp on each update? It should be set once and be the same on every update.

Reply:
The device (hardware) provides a countdown in seconds. If I request a delayed turn off in 120 seconds the API of the hardware returns a decreasing value. I've to transform the countdown to a static timestamp. The could be done once (if I detect a countdown value > 0) but if the users changes the countdown (by the mihome app f.e.) I want to update the timestamp at homeassistant, too (this is an edge-case).

Reply:
In other words: The component provides a cachable/stable timestamp. The timestamp won't be updated if `utcnow()+countdown_value_in_seconds` is in a range of 4 seconds +/- (much space for network latency). If the user changes the `countdown_value_in_seconds` and the difference is bigger then 4 seconds the timestamp (attribute) will be updated.

Reply:
Dang that's gross. Ok.